scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+}
+
+# Check whether process in PID file is still running.
+check_pid_file()
+{
+    local pid_file=$1
+    [ -r "$pid_file" ] && ps -p $(cat $pid_file) >/dev/null 2>&1
+}
+
+# Check client version
+check_client_version()
+{
+    local readonly min_version="8.0.19"
+    IFS="." read -ra min_vers <<< $min_version # split into major minor and patch
+
+    local readonly client_version=${1%%[-/]*} # take only a.b.c from a.b.c-d.e
+    IFS="." read -ra client_vers <<< $client_version
+
+    for i in ${!min_vers[@]}
+    do
+        if [ "${client_vers[$i]}" -lt "${min_vers[$i]}" ]
+        then
+            wsrep_log_error "this operation requires MySQL client version $min_version," \
+                            " this client is '$client_version'"
+            return $EINVAL
+        fi
+    done
+}
+
+trim_to_mysql_user_length()
+{
+    local readonly max_length=32
+    local input=$1
+    local readonly input_length=${#input}
+
+    if [ $input_length -gt $max_length ]
+    then
+        local readonly tail_length=6 # to fit port completely
+        local readonly head_length=$(($max_length - $tail_length))
+        local readonly tail_offset=$(($input_length - $tail_length + 1))
+        local readonly head_str=$(echo $input | cut -c -$head_length)
+        local readonly tail_str=$(echo $input | cut -c $tail_offset-)
+        input=$head_str$tail_str
+    fi
+
+    # here we potentially may want to filter out "bad" characters
+    echo $input
+}
+
+# The following function:
+# 1. checks if CLONE plugin is loaded. If not loads it.
+# 2. checks if SSL cert and key are configured for any of and in that order:
+#    a. CLONE plugin
+#    b. MySQL in general
+#    c. in [sst] section of my.cnf
+# 3. If SSL is configured, but not for the CLONE plugin explicitly, sets up
+#    corresponging CLONE plugin variables
+#
+# Requires environment variables: MYSQL_ACLIENT, MYSQL_PWD
+# Sets the following environment variables:
+# CLEANUP_PLUGIN, REQUIRE_SSL, CLONE_SSL_CERT,
+# CLONE_SSL_KEY, CLONE_SSL_CA, CLIENT_SSL_OPTIONS, CLEANUP_SSL
+#
+setup_clone_plugin()
+{
+    
+    # Either donor or recipient
+    local -r ROLE=$1
+    
+    wsrep_log_debug "-> ############## SSL SECTION [START] ($ROLE)############"
+
+    local CLONE_PLUGIN_LOADED=`$MYSQL_ACLIENT -e "SELECT COUNT(*) FROM INFORMATION_SCHEMA.PLUGINS WHERE PLUGIN_TYPE = 'CLONE';"`
+    
+    wsrep_log_debug "->CLONE_PLUGIN_LOADED: $CLONE_PLUGIN_LOADED"
+    
+    if [ "$CLONE_PLUGIN_LOADED" -eq 0 ]
+    then
+        wsrep_log_info "Installing CLONE plugin"
+
+        # INSTALL PLUGIN is replicated by default, so we need to switch off
+        # session replication on donor
+        if [ "$ROLE" = "donor" ]
+        then
+            WSREP_OFF="SET SESSION wsrep_on=OFF; "
+        else
+            WSREP_OFF="" # joiner does not have replication enabled
+        fi
+        $MYSQL_ACLIENT -e "${WSREP_OFF}INSTALL PLUGIN CLONE SONAME 'mysql_clone.so';"
+        if [ "$ROLE" == "donor" ]; then
```

Comment:
Please address Kamil's comments https://github.com/percona/percona-xtradb-cluster/pull/2018/files#r1939063985
>kamil-holubicki 
>
>It is not consistent across the script how we do the comparison. Sometimes it is == and sometimes it it = >(just 4 lines above).
>= is posix compatible and should be used with single brackets [...]
>== is bash only compatible and should be used with double brackets [[...]]
>
>Would it be possible to have it consistent?

Reply:
as above and duplicate